server/controllers/parcel-controller.js

```diff
@@ -15,64 +38,165 @@ export const getParcel = (req, res) => {
-    }
+    const parcelId = parseInt(req.params.id, 10);
+
+    client.query(`SELECT * FROM parcels WHERE id = ${parcelId};`, (err, resp) => {
+      const parcel = resp.rows[0];
+      if (err) {
+        res.send(err);
+      } else if (!parcel) {
+        res.send({ msg: 'This Parcel Delivery Order Does Not Exist' });
+      } else {
+        const userIdFromToken = parseInt(req.user.userInfo.id, 10);
+        if (userIdFromToken === parcel.user_id) {
+          res.send(parcel);
+        } else {
+          res.send({ msg: 'Sorry you can not fetch Parcel for another User!' });
+        }
+      }
+    });
+
   }
 };
 
-export const updateParcel = (req, res) => {
+export const updateParcelDestination = (req, res) => {
   const errors = validationResult(req);
   if (!errors.isEmpty()) {
     res.status(422).json({ errors: errors.array() });
-  } else {
-    const parcelToCancel = parcelsDb.find(parcel => parcel.id === req.params.id);
-    if (!parcelToCancel) {
-      res.status(404).send({ msg: 'Error! Parcel Order Not Found' });
-    } else if (parcelToCancel.status === 'Delivered') {
-      res.status(400).send({ msg: 'Parcel Delivered! Cannot cancel Parcel Order.' });
-    } else if (parcelToCancel.status === 'Cancelled') {
-      res.status(400).send({ msg: 'Cannot Cancel Parcel Order, Order Already Cancelled.' });
-    } else {
-      parcelToCancel.status = 'Cancelled';
-      fs.writeFile('server/parcelsdb.json', JSON.stringify(parcelsDb, null, 2), (err) => {
-        if (err) {
-          res.status(500).send(err);
+  } else if (req.user.userInfo.role === 'member') {
+    const parcelId = parseInt(req.params.id, 10);
+    client.query(`SELECT * FROM parcels WHERE id = ${parcelId};`, (err, resp) => {
+      const parcel = resp.rows[0];
+      if (err) {
+        res.send(err);
+      } else if (!parcel) {
+        res.send({ msg: 'This Parcel Delivery Order Does Not Exist' });
+      } else {
+        const userIdFromToken = parseInt(req.user.userInfo.id, 10);
+        if (userIdFromToken === parcel.user_id) {
+          if (parcel.status == 'Delivered') {
+            res.send({ msg: 'Sorry, can not Change Destination. Parcel already Delivered' });
+          } else if (parcel.status == 'Cancelled') {
+            res.send({ msg: 'Sorry, can not Change Destination. Parcel already Cancelled' });
+          } else {
+            client.query(`UPDATE parcels SET destination = '${req.body.destination}' WHERE id= ${parcelId} RETURNING *;`, (err, resp) => {
+              if (err) {
+                res.send(err);
+              } else {
+                res.send(resp.rows[0]);
+              }
+            });
+          }
         } else {
-          res.status(202).send(parcelToCancel);
+          res.send({ msg: 'Sorry you can not update Parcel for another User!' });
         }
-      });
-    }
+      }
+    });
+  } else {
+    res.send({ msg: 'Sorry, only users can change the Destination' });
```

Comment:
If you have implemented JWT, you don't need to write this block of code from line 95 to 97.